### PR TITLE
Use docker compose for development

### PR DESCRIPTION
**Pull request by rstefan1**

### CODE REVIEW

**Comment by rstefan1** [comment]
FIXES ENG-336

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+version: "3.3"
+
+services:
+  wordpress:
+    depends_on:
+      - db
+      - memcached
+    image: TODO
```
Comment (rstefan1) [inline docker-compose.yaml:8]: Add the image

File: wp-config-docker-compose.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+$scheme = 'http://';
+if ( 'on' === $_SERVER['HTTPS'] ) {
+    $scheme = 'https://';
+}
+if ( ! defined( 'WP_HOME' ) ) {
+    define( 'WP_HOME', rtrim( getenv( 'WP_HOME' ) ?: $scheme . $_SERVER['HTTP_HOST'], '/' ) );
+}
+if ( ! defined( 'WP_SITEURL' ) ) {
+    define( 'WP_SITEURL', rtrim( getenv( 'WP_SITEURL' ), '/' ) ?: WP_HOME . '/wp' );
```
Comment (AMecea) [inline wp-config-docker-compose.php:95]: Should work without `/wp/` prefix.

```suggestion
    define( 'WP_SITEURL', rtrim( getenv( 'WP_SITEURL' ), '/' ) ?: WP_HOME  );
```

File: wp-config-docker-compose.php [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+ */
+if ( ! defined( 'AUTOMATIC_UPDATER_DISABLED' ) ) {
+    define( 'AUTOMATIC_UPDATER_DISABLED', true );
+}
+if ( ! defined( 'DISABLE_WP_CRON' ) ) {
+    define( 'DISABLE_WP_CRON', getenv( 'DISABLE_WP_CRON' ) ?: false );
+}
+if ( ! defined( 'DISALLOW_FILE_EDIT' ) ) {
+    // Disable the plugin and theme file editor in the admin
+    define( 'DISALLOW_FILE_EDIT', "true" === strtolower( getenv( 'DISALLOW_FILE_EDIT' ) ?: "true" ) );
```
Comment (AMecea) [inline wp-config-docker-compose.php:122]:
```suggestion
    define( 'DISALLOW_FILE_EDIT', "true" === strtolower( getenv( 'DISALLOW_FILE_EDIT' ) ?: "false" ) );
```

File: wp-config-docker-compose.php [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+}
+if ( ! defined( 'DISALLOW_FILE_MODS' ) ) {
+    // Disable plugin and theme updates and installation from the admin
+    define( 'DISALLOW_FILE_MODS', "true" === strtolower( getenv( 'DISALLOW_FILE_MODS' ) ?: "true" ) );
```
Comment (AMecea) [inline wp-config-docker-compose.php:126]:
```suggestion
    define( 'DISALLOW_FILE_MODS', "true" === strtolower( getenv( 'DISALLOW_FILE_MODS' ) ?: "false" ) );
```

**Review by AMecea — CHANGES_REQUESTED** [review]

File: wp-config-docker-compose.php [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    $_SERVER['HTTPS'] = 'on';
+}
+
+$user_config = dirname( __DIR__ ) . '/config/wp-config.php';
+if ( file_exists( $user_config ) ) {
+    require_once $user_config;
+}
+
+$scheme = 'http://';
+if ( 'on' === $_SERVER['HTTPS'] ) {
```
Comment (AMecea) [inline wp-config-docker-compose.php:88]: I see in logs the following warning:
```
[24-Nov-2022 15:39:21 UTC] PHP Warning:  Undefined array key "HTTPS" in /app/web/wp-config.php on line 88
```

Maybe we should  force `http://`.

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    restart: always
+    environment:
+      DB_HOST: db:3306
+      DB_USER: wordpress
+      DB_PASSWORD: not-so-secure
+      DB_NAME: wordpress
+      MEMCACHED_HOST: memcached:11211
+      WP_HOME: http://localhost:8080
+      WP_SITEURL: http://localhost:8080/wp
+      WP_ENV: development
```
Comment (AMecea) [inline docker-compose.yaml:24]: Local network is slower than cloud and 30s of timeout is not enough. 

```suggestion
      WP_ENV: development
      PHP_REQUEST_TIMEOUT: 60

```

**Review by AMecea — CHANGES_REQUESTED** [review]

**Review by cristianuibar — CHANGES_REQUESTED** [review]
Please remove the `provision` folder. It is no longer needed as well as [.gitmodules](https://github.com/presslabs/gitium-base/blob/master/.gitmodules).

In the readme docs please remove the step to apply `git submodule update --init` as that's not required anymore.

Also, please make sure the default NGINX timeout is set at minimum 300s.

**Review by AMecea — CHANGES_REQUESTED** [review]

File: wp-config-docker-compose.php [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+ * a unique prefix. Only numbers, letters, and underscores please!
+ */
+$table_prefix = getenv('DB_PREFIX') ?: 'wp_';
+
+/**
+ * Allow WordPress to detect HTTPS when used behind a reverse proxy or a load balancer
+ * See https://codex.wordpress.org/Function_Reference/is_ssl#Notes
+ */
+if (isset($_SERVER['HTTP_X_FORWARDED_PROTO']) && $_SERVER['HTTP_X_FORWARDED_PROTO'] === 'https') {
+    $_SERVER['HTTPS'] = 'on';
```
[the remark this answers is not in the document]
Comment (AMecea) [reply wp-config-docker-compose.php]: There are some PHP warnings:
```
PHP Warning:  Undefined array key "HTTPS" in /app/web/wp-config.php on line 88
```

**Review by AMecea — CHANGES_REQUESTED** [review]
Please check [this version of docker-compose.yaml](https://github.com/presslabs/gitium-base/commit/82b0e104542a03cffb89120c7fffde81ea1a800c) that works on both linux and mac but on mac requires running with `USER_ID="$(id -u)" GROUP_ID="$(id -g)"`.

File: docker-compose.yaml [hunk 8]
```diff
@@ -0,0 +1,48 @@
+version: "3.3"
+
+services:
+  wordpress:
+    image: docker.io/presslabs/wordpress-runtime:6.1-php-8.0
+    user: "${USER_ID}:${GROUP_ID}"
```
Comment (AMecea) [inline docker-compose.yaml:6]:
```suggestion
    user: "${USER_ID:-1000}:${GROUP_ID:-1000}"

```

**Review by AMecea — CHANGES_REQUESTED** [review]

**Review by AMecea — APPROVED** [review]

File: docker-compose.yaml [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+      - db_data:/var/lib/mysql
+    restart: always
+    environment:
+      MYSQL_ROOT_PASSWORD: not-so-secure
+      MYSQL_DATABASE: wordpress
+      MYSQL_USER: wordpress
+      MYSQL_PASSWORD: not-so-secure
+
+  memcached:
+    image: memcached:1.5
```
Comment (AMecea) [inline docker-compose.yaml:45]:
```suggestion
    image: memcached:1.6
```

**Review by AMecea — CHANGES_REQUESTED** [review]

File: wp-config-docker.php [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+ * @link https://codex.wordpress.org/Editing_wp-config.php
+ *
+ * @package WordPress
+ */
+
+/**
+ * Set up our global environment constant and load its config first
+ * Default: production
+ */
+define('WP_ENV', getenv('WP_ENV') ?: 'production');
```
Comment (AMecea) [inline wp-config-docker.php:25]:
```suggestion
define('WP_ENV', getenv('WP_ENV') ?: 'development');
```

File: wp-config-docker.php [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+define( 'SECURE_AUTH_KEY',  getenv( 'SECURE_AUTH_KEY' ) ?: 'put your unique phrase here' );
+define( 'LOGGED_IN_KEY',    getenv( 'LOGGED_IN_KEY' ) ?: 'put your unique phrase here' );
+define( 'NONCE_KEY',        getenv( 'NONCE_KEY' ) ?: 'put your unique phrase here' );
+define( 'AUTH_SALT',        getenv( 'AUTH_SALT' ) ?: 'put your unique phrase here' );
+define( 'SECURE_AUTH_SALT', getenv( 'SECURE_AUTH_SALT' ) ?: 'put your unique phrase here' );
+define( 'LOGGED_IN_SALT',   getenv( 'LOGGED_IN_SALT' ) ?: 'put your unique phrase here' );
+define( 'NONCE_SALT',       getenv( 'NONCE_SALT' ) ?: 'put your unique phrase here' );
+
+/**#@-*/
+
```
Comment (AMecea) [inline wp-config-docker.php:65]: We need a new category for error reporting. Set by default to output all errors.

```suggestion
/**
 * Error reporting
 **/
 error_reporting(E_ALL);
```

**Review by AMecea — CHANGES_REQUESTED** [review]

File: docker-compose.yaml [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+      - /var/run:uid=${USER_ID:-1000},gid=${GROUP_ID:-1000}
+    environment:
+      WP_ENV: development
+      DB_HOST: db:3306
+      DB_USER: wordpress
+      DB_PASSWORD: not-so-secure
+      DB_NAME: wordpress
+      MEMCACHED_HOST: memcached:11211
+      WP_HOME: http://localhost:8080
+      WP_SITEURL: http://localhost:8080
```
Comment (AMecea) [inline docker-compose.yaml:28]: Let's test without them. Those should not be set.

**Review by AMecea — APPROVED** [review]
LGTM. Please squash commits and merge.

**Comment by rstefan1** [comment]
Change image: `docker.io/presslabs/wordpress-runtime:classic-6.1-php-8.0`

File: docker-compose.yaml [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+      - ./.dev/wp-config-docker.php:/app/web/wp-config.php:rw
+      - ./.dev/php.ini:/usr/local/etc/php/conf.d/zz-local.ini
+    ports:
+      - "8080:8080"
+    restart: always
+    tmpfs:
+      - /usr/local/docker/etc:uid=${USER_ID:-1000},gid=${GROUP_ID:-1000}
+      - /var/run:uid=${USER_ID:-1000},gid=${GROUP_ID:-1000}
+    environment:
+      WP_ENV: development
```
Comment (AMecea) [inline docker-compose.yaml:22]:
```suggestion
      WP_ENVIRONMENT_TYPE: development
```

**Review by AMecea — APPROVED** [review]

**Pull request merged**